pkg/plugin/utils.go

```diff
@@ -36,33 +34,37 @@ const (
 )
 
 func GetIpsByOnField(on string, context Context, phase Phase) (ipList []net.IP, err error) {
-	on = strings.TrimSpace(on)
-	if strings.Contains(on, EqualSymbol) {
-		if (phase != PhasePostInstall && phase != PhasePostJoin) && phase != PhasePreClean {
-			logrus.Warnf("Current phase is %s. When nodes is specified with a label, the plugin action must be PostInstall or PostJoin, ", phase)
-			return nil, nil
+	splits := strings.Split(on, SplitSymbol)
+	for _, split := range splits {
+		var ips []net.IP
+		split = strings.TrimSpace(split)
+		if strings.Contains(split, EqualSymbol) {
+			if (phase != PhasePostInstall && phase != PhasePostJoin) && phase != PhasePreClean {
+				return nil, fmt.Errorf("current phase is %s. When nodes is specified with a label, the plugin action must be PostInstall or PostJoin. ", phase)
+			}
+			client, err := k8s.Newk8sClient()
+			if err != nil {
+				return nil, fmt.Errorf("failed to get k8s client: %v", err)
+			}
+			ips, err = client.ListNodeIPByLabel(strings.TrimSpace(split))
+			if err != nil {
+				return nil, err
+			}
+		} else if split == common.MASTER || split == common.NODE {
```

Comment:
thanks for your great contribution, i am ok with the logic here.  however ,could we use switch case to make the code more readable ?

Reply:
Yep, this change has been updated, take a look?